### PR TITLE
Auto-restart NGINX and use pillars to set conf path when building from source

### PR DESCRIPTION
**Pull request by grobinson-blockchain**

### CODE REVIEW

**Comment by michaelforge** [comment]
If nginx config would containt an error which would block nginx from starting, wouldn't restart feature result in endless restart?

**Comment by grobinson-blockchain** [comment]
When you have an elastic infrastructure (container schedulers, etc) it's relatively typical to use Nixy/consul-template/etc to reconfigure your web server in real time as your scheduler provisions and de-allocates containers.

However, at the same time, you'll often want configuration management to setup a set of initial virtual hosts and symlink them to the sites-enabled directory. The commit above adds support for an option `vhost_overwrite` which lets you dictate whether the NGINX formula can overwrite existing vhosts. It is enabled by default to not break backwards compatibility with existing users.

I hope this is something you'll like to see merged into the formula.

File: nginx/ng/src.sls [hunk 1]
```diff
@@ -29,7 +29,7 @@ nginx_download:
 
 nginx_configure:
   cmd.run:
-    - name: ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf {{ nginx.source.opts | join(' ') }}
+    - name: ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path={{ nginx.lookup.conf_file }} {{ nginx.source.opts | join(' ') }}
```
Comment (gravyboat) [inline nginx/ng/src.sls]: Could you please add `/etc/nginx/nginx.conf` back as the default? Then this will continue to be backwards compatible.

**Comment by gravyboat** [comment]
Other than my one note above this looks good to me.

File: nginx/ng/src.sls [hunk 2]
```diff
@@ -29,7 +29,7 @@ nginx_download:
 
 nginx_configure:
   cmd.run:
-    - name: ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path={{ nginx.lookup.conf_file }} {{ nginx.source.opts | join(' ') }}
+    - name: ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf {{ nginx.source.opts | join(' ') }}
```
Comment (michaelforge) [inline nginx/ng/src.sls]: Thanks!

The thing @gravyboat suggested at #119 is achievable by:
```python
{{ nginx.lookup.conf_file or '/etc/nginx/nginx.conf' }}
```
or
```python
{{ nginx.lookup.conf_file|default('/etc/nginx/nginx.conf') }}
```

**Comment by gravyboat** [comment]
@grobinson-blockchain Apologies for my poor explanation. What @michaelforge is talking about is what I meant!

**Comment by grobinson-blockchain** [comment]
@gravyboat okay, will update!

**Comment by grobinson-blockchain** [comment]
I've added an option to each vhost `overwrite` that lets you declare whether the formula should overwrite the existing vhost file on a vhost by vhost granularity. My previous contribution set this is a blanket option for all vhosts.

If `overwrite` is absent then it will be interpreted as `True` by default to avoid breaking backwards compatibility. Only if `overwrite` is set to false will the formula not overwrite the existing file.

**Comment by gravyboat** [comment]
Good stuff, thanks.

**Pull request merged**